plugins/com.google.cloud.tools.eclipse.appengine.compat.test/src/com/google/cloud/tools/eclipse/appengine/compat/GpeMigratorXsltTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.compat;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class GpeMigratorXsltTest {
+
+  private static final String WTP_METADATA_XML = "<?xml version='1.0' encoding='UTF-8'?>"
+      + "<faceted-project>"
+      + "  <runtime name='Google App Engine'/>"
+      + "  <runtime name='App Engine Standard Runtime'/>"
+      + "  <installed facet='java' version='1.7'/>"
+      + "  <installed facet='jst.web' version='2.5'/>"
+      + "  <installed facet='com.google.appengine.facet' version='1'/>"
+      + "  <installed facet='com.google.appengine.facet.ear' version='1'/>"
+      + "</faceted-project>";
+
+  private static final String STYLESHEET = "<?xml version='1.0' encoding='UTF-8'?>"
+      + "<xsl:stylesheet version='2.0' xmlns:xsl='http://www.w3.org/1999/XSL/Transform'>"
+      + "  <xsl:template match='faceted-project'>"
+      + "    <xsl:copy/>"
+      + "  </xsl:template>"
+      + "</xsl:stylesheet>";
+
+  @Test
+  public void testApplyXslt()
+      throws IOException, ParserConfigurationException, SAXException, TransformerException {
+    try (InputStream xmlStream = stringToInputStream(WTP_METADATA_XML);
+        InputStream stylesheetStream = stringToInputStream(STYLESHEET)) {
+
+      DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+      Document document = builder.parse(xmlStream);
+
+      try (InputStream inputStream = GpeMigrator.applyXslt(document, stylesheetStream)) {
```

Comment:
This API seems strange. Why do we pass a stylesheet to GpeMigrator? Shouldn't this be an internal implementation detail? I.e. why should the caller know or care that GpeMigrator uses XSLT?